examples/custom-node/src/evm.rs

```diff
@@ -23,24 +32,153 @@ use reth_op::{
     node::{
         OpBlockAssembler, OpEvmConfig, OpEvmFactory, OpNextBlockEnvAttributes, OpRethReceiptBuilder,
     },
-    DepositReceipt, OpPrimitives, OpReceipt, OpTransactionSigned,
-};
-use revm::{
-    context::{result::ExecutionResult, TxEnv},
-    database::State,
+    DepositReceipt, OpReceipt, OpTransactionSigned,
 };
+use reth_primitives_traits::NodePrimitives;
+use reth_revm::context::TxEnv;
+use revm::{context::result::ExecutionResult, database::State};
 use std::sync::Arc;
 
+pub struct CustomTxEnv(TxEnv);
```

Comment:
I think we could possible even reuse the existing texenv entire for the payment tx type, but let's do the full example like this